### PR TITLE
Feature/modal/clan

### PR DESCRIPTION
**Pull request by nikita-dnestrov**

### CODE REVIEW

File: client/src/components/modals/clan-modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { Formik, Form, Field } from 'formik';
+import Modal from 'react-modal';
```
Comment (taras-dubyk) [inline client/src/components/modals/clan-modal/index.tsx:3]: Please use our general modal component client/src/components/modals/modal/index.tsx

File: client/src/components/modals/clan-modal/components/image-upload/index.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import styles from './image-upload.module.scss';
+
+interface IImageUploadProps {
+	label: string;
+}
+
+const ImageUpload: React.FC<IImageUploadProps> = (props) => {
+	return (
+		<div className={styles.container}>
+			<img src="https://dess.gov.ua/wp-content/uploads/2020/05/placeholder-1.png" />
```
Comment (taras-dubyk) [inline client/src/components/modals/clan-modal/components/image-upload/index.tsx:11]: Substitute external image with the local one.

File: client/src/components/modals/clan-modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+interface IClanModalProps {
+	token: string;
+}
+
+Modal.setAppElement('#root');
+
+export const ClanModal: React.FC<IClanModalProps> = (props) => {
+	const [modalIsOpen, setIsOpen] = React.useState(false);
+
+	const createClan = React.useCallback((name: string) => {
```
Comment (taras-dubyk) [inline client/src/components/modals/clan-modal/index.tsx:19]: Make call to server inside saga, use `http.callWebApi`

File: client/src/components/modals/clan-modal/mockData.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const token =
```
Comment (taras-dubyk) [inline client/src/components/modals/clan-modal/mockData.ts:1]: remove

**Review by taras-dubyk — CHANGES_REQUESTED** [review]

**Comment by nikita-dnestrov** [comment]
Changed maxwidth and maxheight in general modal component because it cramped content too much

**Review by taras-dubyk — APPROVED** [review]

**Pull request merged**